### PR TITLE
[MIG] account_cutoff_prepaid: Migration to 12.0

### PR DESCRIPTION
**Pull request by jimhoefnagels**
Depends on:

- [ ] #89 
- [ ] #90

### CODE REVIEW

**Comment by jimhoefnagels** [comment]
> Hey @jimhoefnagels, thank you for your Pull Request.
> 
> It looks like some users haven't signed our **C**ontributor **L**icense **A**greement, yet.
> You can read and sign our full Contributor License Agreement here: http://odoo-community.org/page/cla
> Here is a list of the users:
> 
>     * @jimhoefnagels (login unknown in OCA database)
> 
>     * oleksandrpaziuk [o.paziuk@mobilunity.com](mailto:o.paziuk@mobilunity.com) (no github login found)
> 
> 
> Appreciation of efforts,
> OCA CLAbot

CLA has been signed and sent for @jimhoefnagels on Mon, 17 Dec 2018

kind regards
Jim

**Comment by jimhoefnagels** [comment]
Depends on PR89 and 90

File: account_cutoff_prepaid/readme/CONTRIBUTORS.rst [hunk 1]
```diff
@@ -0,0 +1,3 @@
+* Alexis de Lattre <alexis.delattre@akretion.com>
+* Stéphane Bidoul <stephane.bidoul@acsone.eu>
+* Jim Hoefnagels <jim.hoefnagels@dynapps.be>
```
Comment (rven) [inline account_cutoff_prepaid/readme/CONTRIBUTORS.rst]: Missing newline (missing-newline-extrafiles)
```suggestion
* Jim Hoefnagels <jim.hoefnagels@dynapps.be>

```

File: account_cutoff_prepaid/readme/CONFIGURE.rst [hunk 2]
```diff
@@ -0,0 +1,5 @@
+On the form view of the company, in the *Configuration* tab, there is a *Cut-offs* section where you can configure:
+
+* the *Default Cut-off Journal*, which is the journal in which the cut-off journal items will be generated by default,
+* the *Default Account for Prepaid Revenue*,
+* the *Default Account for Prepaid Expense*.
```
Comment (rven) [inline account_cutoff_prepaid/readme/CONFIGURE.rst]: Missing newline (missing-newline-extrafiles)
```suggestion
* the *Default Account for Prepaid Expense*.

```

File: account_cutoff_prepaid/models/account_cutoff.py [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+                res = src_journals.ids
+        return res
+
+    source_journal_ids = fields.Many2many(
+        'account.journal', column1='cutoff_id', column2='journal_id',
+        string='Source Journals', readonly=True,
+        default=lambda self: self._get_default_source_journals(),
+        states={'draft': [('readonly', False)]})
+    forecast = fields.Boolean(
+        string='Forecast',
```
Comment (rven) [inline account_cutoff_prepaid/models/account_cutoff.py]: The attribute string is redundant. String parameter equal to name of variable (attribute-string-redundant)
```suggestion
```

File: account_cutoff_prepaid/models/account_cutoff.py [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+        readonly=True, states={'draft': [('readonly', False)]},
+        help="The Forecast mode allows the user to compute "
+        "the prepaid revenue/expense between 2 dates in the future.")
+    start_date = fields.Date(string='Start Date')
```
Comment (rven) [inline account_cutoff_prepaid/models/account_cutoff.py]: The attribute string is redundant. String parameter equal to name of variable (attribute-string-redundant)
```suggestion
    start_date = fields.Date()
```

File: account_cutoff_prepaid/models/account_cutoff.py [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+    end_date = fields.Date(string='End Date')
```
Comment (rven) [inline account_cutoff_prepaid/models/account_cutoff.py]: The attribute string is redundant. String parameter equal to name of variable (attribute-string-redundant)
```suggestion
    end_date = fields.Date()
```

File: account_cutoff_prepaid/models/account_cutoff.py [hunk 6]
```diff
@@ -0,0 +1,192 @@
...
+class AccountCutoffLine(models.Model):
+    _inherit = 'account.cutoff.line'
+
+    move_line_id = fields.Many2one(
+        'account.move.line', string='Account Move Line', readonly=True)
+    move_date = fields.Date(
+        related='move_line_id.date', string='Account Move Date', readonly=True)
+    invoice_id = fields.Many2one(
+        related='move_line_id.invoice_id', string='Invoice', readonly=True)
+    start_date = fields.Date(string='Start Date', readonly=True)
```
Comment (rven) [inline account_cutoff_prepaid/models/account_cutoff.py]: The attribute string is redundant. String parameter equal to name of variable (attribute-string-redundant)
```suggestion
    start_date = fields.Date(readonly=True)
```

File: account_cutoff_prepaid/models/account_cutoff.py [hunk 7]
```diff
@@ -0,0 +1,192 @@
...
+    end_date = fields.Date(string='End Date', readonly=True)
```
Comment (rven) [inline account_cutoff_prepaid/models/account_cutoff.py]: The attribute string is redundant. String parameter equal to name of variable (attribute-string-redundant)
```suggestion
    end_date = fields.Date(readonly=True)
```

File: account_cutoff_prepaid/models/account_cutoff.py [hunk 8]
```diff
@@ -0,0 +1,192 @@
...
+    total_days = fields.Integer('Total Number of Days', readonly=True)
+    prepaid_days = fields.Integer(
+        string='Prepaid Days', readonly=True,
```
Comment (rven) [inline account_cutoff_prepaid/models/account_cutoff.py]: The attribute string is redundant. String parameter equal to name of variable (attribute-string-redundant)
```suggestion
        readonly=True,
```

**Review by rven — CHANGES_REQUESTED** [review]
Small code changes are needed to be conform the OCA requirements.

**Pull request merged**